File: src/test/java/no/nav/sosialhjelp/soknad/business/service/informasjon/PabegynteSoknaderServiceTest.java
package no.nav.sosialhjelp.soknad.business.service.informasjon;

import no.nav.sosialhjelp.soknad.business.db.repositories.soknadmetadata.SoknadMetadataRepository;
import no.nav.sosialhjelp.soknad.business.domain.SoknadMetadata;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

import static java.util.Collections.emptyList;
import static java.util.Collections.singletonList;
import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.ArgumentMatchers.anyString;
import static org.mockito.Mockito.when;

@ExtendWith(MockitoExtension.class)
class PabegynteSoknaderServiceTest {

    @Mock
    private SoknadMetadataRepository soknadMetadataRepository;

    @InjectMocks
    private PabegynteSoknaderService pabegynteSoknaderService;

    @Test
    void brukerHarIngenPabegynteSoknader() {
        when(soknadMetadataRepository.hentPabegynteSoknaderForBruker(anyString())).thenReturn(emptyList());

        assertThat(pabegynteSoknaderService.hentPabegynteSoknaderForBruker("fnr")).isEmpty();
    }

    @Test
    void brukerHar1PabegyntSoknad() {
        var now = LocalDateTime.now();
        var soknadMetadata = new SoknadMetadata();
        soknadMetadata.sistEndretDato = now;
        soknadMetadata.behandlingsId = "id";

        when(soknadMetadataRepository.hentPabegynteSoknaderForBruker(anyString())).thenReturn(singletonList(soknadMetadata));

        var pabegyntSoknadList = pabegynteSoknaderService.hentPabegynteSoknaderForBruker("fnr");
        assertThat(pabegyntSoknadList).hasSize(1);
        assertThat(pabegyntSoknadList.get(0).getSistOppdatert()).isEqualTo(now.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME));
        assertThat(pabegyntSoknadList.get(0).getBehandlingsId()).isEqualTo("id");
    }
}

File: src/test/kotlin/no/nav/sosialhjelp/soknad/api/informasjon/PabegynteSoknaderServiceTest.kt
package no.nav.sosialhjelp.soknad.api.informasjon

import io.mockk.every
import io.mockk.mockk
import no.nav.sosialhjelp.soknad.v2.metadata.SoknadMetadataService
import no.nav.sosialhjelp.soknad.v2.metadata.SoknadType
import no.nav.sosialhjelp.soknad.v2.metadata.Tidspunkt
import no.nav.sosialhjelp.soknad.v2.soknad.SoknadService
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.time.LocalDateTime
import java.util.UUID

internal class PabegynteSoknaderServiceTest {
    private val soknadMetadataService: SoknadMetadataService = mockk()
    private val soknadService: SoknadService = mockk()

    private val pabegynteSoknaderService = PabegynteSoknaderService(soknadMetadataService, soknadService)

    @Test
    fun brukerHarIngenPabegynteSoknader() {
        every { soknadService.findOpenSoknadIds(any()) } returns emptyList()
        every { soknadMetadataService.getMetadatasForIds(any()) } returns emptyList()

        assertThat(pabegynteSoknaderService.hentPabegynteSoknaderForBruker("fnr")).isEmpty()
    }

    @Test
    fun brukerHar1PabegyntSoknad() {
        val now = LocalDateTime.now()
        val soknadId = UUID.randomUUID()
        val soknadMetadata =
            no.nav.sosialhjelp.soknad.v2.metadata.SoknadMetadata(
                soknadId = soknadId,
                personId = "fnr",
                tidspunkt =
                    Tidspunkt(
                        opprettet = now,
                        sistEndret = now,
                        sendtInn = now,
                    ),
                soknadType = SoknadType.STANDARD,
            )
        every { soknadService.findOpenSoknadIds(any()) } returns listOf(soknadMetadata.soknadId)
        every { soknadMetadataService.getMetadatasForIds(listOf(soknadMetadata.soknadId)) } returns listOf(soknadMetadata)

        val pabegyntSoknadList = pabegynteSoknaderService.hentPabegynteSoknaderForBruker("fnr")

        assertThat(pabegyntSoknadList).hasSize(1)
        assertThat(pabegyntSoknadList[0].behandlingsId).isEqualTo(soknadId.toString())
    }
}
